packages/block-library/src/navigation/style.scss

```diff
@@ -26,8 +26,9 @@
 	}
 
 	// Menu item link.
-	.wp-block-pages-list__item__link,
-	.wp-block-navigation-link__content {
+	// The addition of :not(.wp-block-button_link) is necessary to override generic link colors from global styles.
+	.wp-block-pages-list__item__link:not(.wp-block-button_link),
+	.wp-block-navigation-link__content:not(.wp-block-button_link) {
```

Comment:
I could have added `:not(.any-random-class)` and it would have worked. That's why I'm not really a fan of using :not in the first place. It's essentially an "allow-list" of blocks to ignore.